File: src/python_example/classification/fastvit_ma36/fastvit_ma36_async.py
#!/usr/bin/env python3
# Copyright (C) 2018- DEEPX Ltd. All rights reserved.
"""
Fastvit ma36 Asynchronous Inference Example

Usage:
    python fastvit_ma36_async.py --model model.dxnn --image input.jpg
"""

import sys
from pathlib import Path

_module_dir = Path(__file__).parent
_v3_dir = _module_dir.parent.parent
for _path in [str(_v3_dir), str(_module_dir)]:
    if _path not in sys.path:
        sys.path.insert(0, _path)

from factory import Fastvit_ma36Factory
from common.runner import AsyncRunner, parse_common_args


def parse_args():
    return parse_common_args("Fastvit ma36 Sync Inference")


def main():
    args = parse_args()
    factory = Fastvit_ma36Factory()
    runner = AsyncRunner(factory)
    runner.run(args)


if __name__ == "__main__":
    main()

File: src/cpp_example/classification/fastvit_ma36/fastvit_ma36_async.cpp
/**
 * @file fastvit_ma36_async.cpp
 * @brief FastvitMa36Factory asynchronous inference example
 */

#include "factory/fastvit_ma36_factory.hpp"
#include "common/runner/async_classification_runner.hpp"

int main(int argc, char* argv[]) {
    auto factory = std::make_unique<dxapp::FastvitMa36Factory>();
    dxapp::AsyncClassificationRunner<dxapp::FastvitMa36Factory> runner(std::move(factory));
    return runner.run(argc, argv);
}
